sdk/src/client/builder.rs

```diff
@@ -416,3 +357,110 @@ impl ClientBuilder {
         }
     }
 }
+
+pub mod dto {
+    use serde::{Deserialize, Serialize};
+
+    use super::*;
+
+    #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+    #[serde(rename_all = "camelCase")]
+    pub struct ClientBuilderDto {
+        /// Node manager builder
+        #[serde(flatten)]
+        pub node_manager_builder: crate::client::node_manager::builder::NodeManagerBuilder,
+        /// Options for the MQTT broker
+        #[cfg(feature = "mqtt")]
+        #[cfg_attr(docsrs, doc(cfg(feature = "mqtt")))]
+        #[serde(flatten)]
+        pub broker_options: BrokerOptions,
+        /// Data related to the used network
+        #[serde(flatten, default)]
+        pub network_info: NetworkInfoDto,
+        /// Timeout for API requests
+        #[serde(default = "default_api_timeout")]
+        pub api_timeout: Duration,
+        /// Timeout when sending a block that requires remote proof of work
+        #[serde(default = "default_remote_pow_timeout")]
+        pub remote_pow_timeout: Duration,
+        /// The amount of threads to be used for proof of work
+        #[serde(default)]
+        #[cfg(not(target_family = "wasm"))]
+        pub pow_worker_count: Option<usize>,
+    }
+
+    impl TryFrom<ClientBuilderDto> for ClientBuilder {
+        type Error = crate::client::Error;
+
+        fn try_from(value: ClientBuilderDto) -> core::result::Result<Self, Self::Error> {
+            Ok(Self {
+                node_manager_builder: value.node_manager_builder,
+                #[cfg(feature = "mqtt")]
+                broker_options: value.broker_options,
+                network_info: value.network_info.try_into()?,
+                api_timeout: value.api_timeout,
+                remote_pow_timeout: value.remote_pow_timeout,
+                #[cfg(not(target_family = "wasm"))]
+                pow_worker_count: value.pow_worker_count,
+            })
+        }
+    }
+
+    impl From<&ClientBuilder> for ClientBuilderDto {
+        fn from(value: &ClientBuilder) -> Self {
+            Self {
+                node_manager_builder: value.node_manager_builder.clone(),
+                #[cfg(feature = "mqtt")]
+                broker_options: value.broker_options,
+                network_info: (&value.network_info).into(),
+                api_timeout: value.api_timeout,
+                remote_pow_timeout: value.remote_pow_timeout,
+                #[cfg(not(target_family = "wasm"))]
+                pow_worker_count: value.pow_worker_count,
+            }
+        }
+    }
+
+    /// Dto for the NetworkInfo
+    #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
```

Comment:
alright Alex ... you not gonna like this comment ... but ... your `derive` order is inconsistent. :hugs:

Reply:
whatever...PR lgtm.